aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -831,8 +831,6 @@ TORCH_IMPL_FUNC(index_add_cpu_out)
   auto index_contig = index.contiguous();
 
   if (source.dim() != 0) {
-    TORCH_CHECK(source.dim() == numel, "source tensor shape along dim does not match index tensor length.");
```

Comment:
Perhaps you meant `source.size(dim) == numel`?

Reply:
Also, try running the tests locally before pushing when possible to save on CI runs.

Reply:
Yes 100%, I'm not sure how my initial changes were erased -- I still thought it might make sense to remove it as the same check is done on line ~335 in index_func_meta_impl.
```
TORCH_CHECK(numel == (source.dim() == 0 ? 1 : source.size(dim)),
              func, "_(): Number of indices (", numel, ") should be equal to source.size(dim): (",
              source.size(dim), "), for dim: ", dim)
```
Do you think I should reinclude the removed check?

Reply:
Then, yeah, don't add this check. But also, add the checks in `index_func_meta_impl` that way we can avoid the repetition.

Reply:
> Then, yeah, don't add this check. But also, add the checks in `index_func_meta_impl` that way we can avoid the repetition.

That makes sense, after checking the documentation it seems like [reduce](https://pytorch.org/docs/stable/generated/torch.Tensor.index_reduce_.html)/[select](https://pytorch.org/docs/stable/generated/torch.index_select.html)/etc require the same.

Reply:
you should have a look at which tests are failing and then run `pytest test/whichever_file_is_the_test_in.py -vk name_of_test`